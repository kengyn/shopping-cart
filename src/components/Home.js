import React from "react";
import { useNavigate } from "react-router-dom";
const Home = () => {
  let navigate = useNavigate();
  return (
    <>
      <h1>Home</h1>

      <button onClick={() => navigate("shop")}>SHOP NOW</button>
    </>
  );
};

export default Home;
