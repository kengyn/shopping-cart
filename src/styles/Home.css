main {
  display: flex;
  width: 80vw;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
}
.main-content {
  position: relative;
}
main img {
  margin-top: 5rem;
}
.main-content button {
  position: absolute;
  top: 84%;
  left: 80%;
  padding: 10px 10px;
  background-color: white;
  border: solid 3px black;
  border-radius: 5px;
  font-size: 0.9rem;
  font-weight: bold;
  cursor: pointer;
}

.main-content button:hover {
  background-color: black;
  color: white;
}
