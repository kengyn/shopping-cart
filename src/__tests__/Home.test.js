import React from "react";
import { render } from "@testing-library/react";
import Home from "../pages/Home";
import "@testing-library/jest-dom";
import { MemoryRouter as Router } from "react-router-dom";

describe("Home component", () => {
  it("renders shop button", () => {
    const { getByRole } = render(
      <Router>
        <Home />
      </Router>
    );
    expect(getByRole("button").textContent).toMatch(/sHop NoW/i);
  });

  it("renders image", () => {
    const { container } = render(
      <Router>
        <Home />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
