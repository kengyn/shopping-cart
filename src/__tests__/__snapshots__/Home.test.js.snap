// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders image 1`] = `
<div>
  <main>
    <div
      class="main-content"
    >
      <img
        alt="gachagacha"
        src="GACHAAA.png"
      />
      <button>
        SHOP NOW
      </button>
    </div>
  </main>
</div>
`;
